inttestutils/transfer.go

```diff
@@ -0,0 +1,240 @@
+		if err != nil {
+			return err
+		}
+	}
+	return errors.New("timeout occurred during attempt to refresh repositories in source instance")
+}
+
+// Return the number of artifacts in the given pattern
+// pattern - Search wildcard pattern
+// serverDetails - The Artifactory server details
+// t - The testing object
+func CountArtifactsInPath(pattern string, serverDetails *config.ServerDetails, t *testing.T) int {
+	searchCmd := generic.NewSearchCommand()
+	searchCmd.SetServerDetails(serverDetails).SetSpec(spec.NewBuilder().Pattern(pattern).BuildSpec())
+	reader, err := searchCmd.Search()
+	assert.NoError(t, err)
+	defer assert.NoError(t, reader.Close())
+	lenght, err := reader.Length()
+	assert.NoError(t, err)
+	return lenght
+}
+
+// Wait for a metadata file to automatically generated in Artifactory
+// pattern - The search pattern
+// serverDetails - The Artifactory server details
+// t - The testing object
+func WaitForCreationInArtifactory(pattern string, serverDetails *config.ServerDetails, t *testing.T) {
+	searchCmd := generic.NewSearchCommand()
+	searchCmd.SetServerDetails(serverDetails).SetSpec(spec.NewBuilder().Pattern(pattern).BuildSpec())
+	for i := 0; i < 20; i++ {
+		reader, err := searchCmd.Search()
+		assert.NoError(t, err)
+		defer assert.NoError(t, reader.Close())
+		if !reader.IsEmpty() {
+			return
+		}
+		time.Sleep(5 * time.Second)
+	}
+	assert.Fail(t, "Couldn't find in target Artifactory: "+pattern)
+}
+
+// Asynchronously execute transfer-files
+// artifactoryCli - Source Artifactory CLI
+// wg - Wait group to update when done
+// filesTransferFinished - Changes to true when the file transfer process finished
+// t - The testing object
+func AsyncExecTransferFiles(artifactoryCli *tests.JfrogCli, wg *sync.WaitGroup, filesTransferFinished *bool, t *testing.T) {
+	wg.Add(1)
+	go func() {
+		defer func() {
+			wg.Done()
+			*filesTransferFinished = true
+		}()
+		// Execute transfer-files
+		assert.NoError(t, artifactoryCli.WithoutCredentials().Exec("transfer-files", "default", "target", "--include-repos="+tests.RtRepo1))
+	}()
+}
+
+// Asynchronously update the number of threads in transfer-files command
+// wg - Wait group to update when done
+// filesTransferFinished - Changes to true when the file transfer process finished
+// t - The testing object
+func AsyncUpdateThreads(wg *sync.WaitGroup, filesTransferFinished *bool, t *testing.T) {
+	wg.Add(1)
+	go func() {
+		threadsCount := 0
+		defer wg.Done()
+
+		// Wait for the number of threads to be updated to the non-zero default and increace the number of threads by 1
+		for !*filesTransferFinished {
+			threadsCount = transferfiles.GetThreads()
+			if threadsCount == 0 {
+				time.Sleep(time.Second)
+				continue
+			}
+			conf := &utils.TransferSettings{ThreadsNumber: threadsCount + 1}
+			assert.NoError(t, utils.SaveTransferSettings(conf))
+			break
+		}
+
+		// Wait for the number of threads to be increase by 1
+		threadCountChanged := false
```

Comment:
I think this variable is redundant. We can just return when the threads count change (assert correct number), and `assert.Fail(t, "number of threads did not change as expected...")` at the end of the function.